### PR TITLE
November Changelog

### PR DESCRIPTION
**Pull request by filiptronicek**
An automatic update to the changelog for November

### [`Live Preview`](https://deploy-preview-3044--gitpod-kumquat.netlify.app/changelog/november-changelog)

---

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3044"></a>

### CODE REVIEW

**Comment by filiptronicek** [comment]
[Preview deployment](https://deploy-preview-3044--gitpod-kumquat.netlify.app/changelog/)

**Comment by loujaybee** [comment]
@filiptronicek - I can't see the preview of the November changelog page in preview or Gitpod env, do you?

**Comment by filiptronicek** [comment]
Thanks for sharing your issue with previewing, @loujaybee, I was silly and used the new folder structure for this changelog, even though the changes are not yet merged. Just fixed this with https://github.com/gitpod-io/website/pull/3044/commits/5dae0a16f69c3a191b25871b6827919709cede87.

**Comment by Siddhant-K-code** [comment]
> What is the slug/URL for the changelog page?

Here is the live preview (also added at the top of this PR convo.):

### [`Live Preview`](https://deploy-preview-3044--gitpod-kumquat.netlify.app/changelog/november-changelog)

File: src/lib/contents/changelog/2022-11-30.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+alt: 2022-11-30 image
+---
+
+<script>
+  import Contributors from "$lib/components/changelog/contributors.svelte";
+</script>
+
+TODO: Insert the featured changelog entry here.
+
+<p><Contributors usernames="" /></p>
```
Comment (Siddhant-K-code) [inline src/lib/contents/changelog/2022-11-30.md:15]: Although PR is in draft. This seems to render like this, right now: 


image


Are we planning to list all contributors (part of this month's releases) here?

**Suggestion**: we can run a follow-up function in the script which would parse through the contributors to the list here

Comment (loujaybee) [reply src/lib/contents/changelog/2022-11-30.md]: Thanks, @Siddhant-K-code !

- Updated the intro
- Added TOC
- Fixed broken headers

Should be all good now!

image

Comment (loujaybee) [reply src/lib/contents/changelog/2022-11-30.md]: > Are we planning to list all contributors (part of this month's releases) here? Suggestion: we can run a follow-up function in the script which would parse through the contributors to the list here

They currently do show per PR—showing everyone at the top doesn't make much sense IMO. 

Happy to discuss though, CC: @filiptronicek for the feedback. 

image

File: src/lib/contents/changelog/2022-11-30.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+A common question we get from Gitpod users is "_how can I configure JetBrains IDE settings so that they're saved between workspace starts_". Without some way to sync IDE settings, you are missing out on fully embracing the power of Gitpod, ephemeral workspaces and [cloud development environments](https://www.gitpod.io/cde).
+
+You can now see how to sync your IDE settings both for your [Project](/docs/configure/projects), using your [Workspace Image](/docs/configure/workspaces/workspace-image) or via committing settings to Git. You can even set settings on a per user basis by leveraging [Dotfiles](/docs/configure/user-settings/dotfiles).
+
+For example, see [IntelliJ IDEA](/docs/references/ides-and-editors/intellij) for how to configure IDE settings syncing.
+
+See your favorite JetBrains [IDE page](/docs/references/ides-and-editors) for more.
+
+## JetBrains Maven Improvements
```
Comment (loujaybee) [inline src/lib/contents/changelog/2022-11-30.md:54]: @felladrin - Are you able to draft up some notes for what has changed here, then I can clean up?

File: src/lib/contents/changelog/2022-11-30.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+**Getting started with CLion or Rider:**
+
+1. **Install [JetBrains Gateway](https://www.jetbrains.com/help/idea/remote-development-a.html#gateway)** - With the [JetBrains Gateway and Gitpod](/docs/integrations/jetbrains-gateway) plugin you can create and manage your latest 20 Gitpod workspaces.
+2. **Install the Gitpod plugin** - Open JetBrains Gateway and you'll see the Gitpod logo on the main page. Click "install" to install the [Gitpod plugin](https://plugins.jetbrains.com/plugin/18438-gitpod-gateway) for JetBrains Gateway.
+3. **Update your Gitpod preferences** - Select CLion or Rider on the [Gitpod preferences page](https://gitpod.io/preferences) which will set the IDE as your default for future workspace starts.
+4. **Start (or restart) your workspace** - Either start a workspace directly from within [JetBrains Gateway](/docs/integrations/jetbrains-gateway) via the Gitpod plugin **OR** open a new workspace directly in Gitpod where on workspace start you will be prompted to open CLion or Rider for that workspace.
+
+## Sync JetBrains IDE Settings between restarts
+
+<!-- TODO: We need a visual here -->
```
Comment (loujaybee) [inline src/lib/contents/changelog/2022-11-30.md:44]: Nice to have a visual here, but not sure what is best, or if anything really showcases this 🤔 

CC: @akosyakov, @andreafalzetti, @felladrin

File: src/lib/contents/changelog/2022-11-30.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+- [CLion and Rider are now out in beta! 🎉](#clion-and-rider-are-now-out-in-beta-)
+- [Sync JetBrains IDE Settings between restarts](#sync-jetbrains-ide-settings-between-restarts)
+- [JetBrains Maven Improvements](#jetbrains-maven-improvements)
+- [Reliable VS Code extensions: Gitpod now hosting an Open VSX Mirror!](#reliable-vs-code-extensions-gitpod-now-hosting-an-open-vsx-mirror)
+- [Manage workspace ports without needing to leave your JetBrains UI](#manage-workspace-ports-without-needing-to-leave-your-jetbrains-ui)
+- [Dashboard](#dashboard)
+- [JetBrains](#jetbrains)
+- [Gitpod CLI](#gitpod-cli)
+- [Workspace](#workspace)
+- [Fixes and improvements](#fixes-and-improvements)
```
Comment (loujaybee) [inline src/lib/contents/changelog/2022-11-30.md:26]:
```suggestion
```

File: src/lib/contents/changelog/2022-11-30.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+- [JetBrains Maven Improvements](#jetbrains-maven-improvements)
+- [Reliable VS Code extensions: Gitpod now hosting an Open VSX Mirror!](#reliable-vs-code-extensions-gitpod-now-hosting-an-open-vsx-mirror)
+- [Manage workspace ports without needing to leave your JetBrains UI](#manage-workspace-ports-without-needing-to-leave-your-jetbrains-ui)
+
+## CLion and Rider are now out in beta! 🎉
+
+1. First we announced [IntelliJ IDEA, PyCharm, GoLand, and PhpStorm](https://www.gitpod.io/blog/gitpod-jetbrains)
+2. Then it was the turn of [RubyMine and WebStorm](https://www.gitpod.io/blog/webstorm-rubymine-now-in-beta)
+
+Today—as part of our [official partnership with JetBrains](https://www.gitpod.io/blog/gitpod-jetbrains)—we "complete the set" of JetBrains IDEs that integrate with Gitpod via [JetBrains Gateway](/docs/integrations/jetbrains-gateway). Gitpod now has first-class IDE integration support for both RubyMine and CLion! 🤘
```
Comment (akosyakov) [inline src/lib/contents/changelog/2022-11-30.md:28]: RubyMine -> Rider

File: src/lib/contents/changelog/2022-11-30.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+**Getting started with CLion or Rider:**
+
+1. **Install [JetBrains Gateway](https://www.jetbrains.com/help/idea/remote-development-a.html#gateway)** - With the [JetBrains Gateway and Gitpod](/docs/integrations/jetbrains-gateway) plugin you can create and manage your latest 20 Gitpod workspaces.
+2. **Install the Gitpod plugin** - Open JetBrains Gateway and you'll see the Gitpod logo on the main page. Click "install" to install the [Gitpod plugin](https://plugins.jetbrains.com/plugin/18438-gitpod-gateway) for JetBrains Gateway.
+3. **Update your Gitpod preferences** - Select CLion or Rider on the [Gitpod preferences page](https://gitpod.io/preferences) which will set the IDE as your default for future workspace starts.
+4. **Start (or restart) your workspace** - Either start a workspace directly from within [JetBrains Gateway](/docs/integrations/jetbrains-gateway) via the Gitpod plugin **OR** open a new workspace directly in Gitpod where on workspace start you will be prompted to open CLion or Rider for that workspace.
+
+## Sync JetBrains IDE Settings between restarts
```
Comment (akosyakov) [inline src/lib/contents/changelog/2022-11-30.md:37]: it is not about  preserving between restarts, settings were already preserving between restarts. It is nota bout synching either.

It is about ability to configure global IDE settings (not project, module level), for repo, image or a user

File: src/lib/contents/changelog/2022-11-30.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+See your favorite JetBrains [IDE page](/docs/references/ides-and-editors) for more.
+
+## JetBrains Maven Improvements
+
+- Projects using prebuilds and Maven, will load successfully and it will mak\*e use of pre-built indexes.
+- Improves the user experience and reduces the time a user has to wait for the IDE to index the project.
+
+<!-- TODO -->
+
+## Reliable VS Code extensions: Gitpod now hosting an Open VSX Mirror!
```
Comment (akosyakov) [inline src/lib/contents/changelog/2022-11-30.md:56]: 🎉

File: src/lib/contents/changelog/2022-11-30.md (hunk 5 above)
Comment (felladrin) [reply src/lib/contents/changelog/2022-11-30.md:28]:
```suggestion
Today—as part of our [official partnership with JetBrains](https://www.gitpod.io/blog/gitpod-jetbrains)—we "complete the set" of JetBrains IDEs that integrate with Gitpod via [JetBrains Gateway](/docs/integrations/jetbrains-gateway). Gitpod now has first-class IDE integration support for both Rider and CLion! 🤘
```

File: src/lib/contents/changelog/2022-11-30.md (hunk 2 above)
Comment (felladrin) [reply src/lib/contents/changelog/2022-11-30.md:49]: Maybe we don't need to mention it, cause it wasn't so impacting:

We've fixed issues causing the workspaces not to start correctly and preventing Run Configurations from running appropriately when a repository containing Maven Wrapper was being pre-built.

**Comment by andreafalzetti** [comment]
@filiptronicek how can I re-generate the automated section?

**Comment by filiptronicek** [comment]
@andreafalzetti currently it's unfortunately kinda hard. You have to do it in another workspace on https://github.com/gitpod-io/website/tree/ft/auto-changelog-prs with `npm run generate:changelog -- --dryRun --onlyPrs`, because we're waiting on the merge of https://github.com/gitpod-io/website/pull/2926.

**Comment by andreafalzetti** [comment]
@filiptronicek do you suggest to run it as you said, and I can safely manually replace the section without risk of being overwritten?

**Comment by filiptronicek** [comment]
@andreafalzetti you can safely copy the output of the command and replace the autogenerated block (`BEGIN_AUTOGENERATED_CHANGES`). This block should be overwritten on every update, as opposed to the rest of the changelog doc.

File: src/lib/contents/changelog/2022-11-30.md (hunk 3 above)
Comment (andreafalzetti) [reply src/lib/contents/changelog/2022-11-30.md:39]: this change is hard to capture in one or two screens

File: src/lib/contents/changelog/2022-11-30.md (hunk 2 above)
Comment (andreafalzetti) [reply src/lib/contents/changelog/2022-11-30.md:49]: we agreed to remove the block about Maven, and keep the commit below

File: src/lib/contents/changelog/2022-11-30.md (hunk 6 above)
Comment (andreafalzetti) [reply src/lib/contents/changelog/2022-11-30.md:37]: copy was updated, @akosyakov feel free to check the new one and leave a new review if you can

File: src/lib/contents/changelog/2022-11-30.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+## Reduced IDE startup time & Custom Shells [Breaking Change]
+
+If you don't use custom shells such as `zsh` or `fish`, you just need to know that we made some changes to speed up the IDE startup time! 🎉
+
+If you use a custom shell, and you rely on bash profile to configure your workspace, continue reading..
+
+In an effort to speed-up the IDE startup time, we made some changes to how we launch IDEs inside workspaces. Previously, we would automatically source interactive login bash shell's profile, but it was a side-effect of our method of launching the IDE. We now changed this behaviur and Gitpod uses the shell configured by the user via the `SHELL` env var. If you use a custom shell but you're relying on configurations specified in bash profile, you might need to update your configuration, check out [Configure a custom shell](/docs/configure/workspaces/workspace-image#workspace-image) for examples.
```
Comment (akosyakov) [inline src/lib/contents/changelog/2022-11-30.md:73]: @andreafalzetti a link here should point to `/docs/configure/workspaces/workspace-image#configure-a-custom-shell`

Comment (andreafalzetti) [reply src/lib/contents/changelog/2022-11-30.md]: good catch, I was sure I've already linked there 😄

**Review by atduarte — APPROVED** [review]
🚀

File: src/lib/contents/changelog/2022-11-30.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+2. **Install the Gitpod plugin** - Open JetBrains Gateway and you'll see the Gitpod logo on the main page. Click "install" to install the [Gitpod plugin](https://plugins.jetbrains.com/plugin/18438-gitpod-gateway) for JetBrains Gateway.
+3. **Update your Gitpod preferences** - Select CLion or Rider on the [Gitpod preferences page](https://gitpod.io/preferences) which will set the IDE as your default for future workspace starts.
+4. **Start (or restart) your workspace** - Either start a workspace directly from within [JetBrains Gateway](/docs/integrations/jetbrains-gateway) via the Gitpod plugin **OR** open a new workspace directly in Gitpod where on workspace start you will be prompted to open CLion or Rider for that workspace.
+
+## Configure JetBrains IDE Settings across workspaces
+
+You can now configure your IDE settings both for your [Project](/docs/configure/projects), using your [Workspace Image](/docs/configure/workspaces/workspace-image) or via committing settings to Git. You can even set settings on a per user basis by leveraging [Dotfiles](/docs/configure/user-settings/dotfiles).
```
Comment (lucasvaltl) [inline src/lib/contents/changelog/2022-11-30.md:40]:
```suggestion
You can now configure your IDE settings at the project and user level: at the your [Project](/docs/configure/projects) level, you can use your [Workspace Image](/docs/configure/workspaces/workspace-image) or commit settings to Git. At the user level, you can now configure settings by leveraging [Dotfiles](/docs/configure/user-settings/dotfiles).
```

File: src/lib/contents/changelog/2022-11-30.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+A common question we get from Gitpod users is "_how can I configure JetBrains IDE settings so that they're saved across workspaces_". Without some way to sync IDE settings, you are missing out on fully embracing the power of Gitpod, ephemeral workspaces and [cloud development environments](https://www.gitpod.io/cde).
+
+For example, see [IntelliJ IDEA](/docs/references/ides-and-editors/intellij) for how to configure IDE settings.
```
Comment (lucasvaltl) [inline src/lib/contents/changelog/2022-11-30.md:44]:
```suggestion
For an example, see [IntelliJ IDEA](/docs/references/ides-and-editors/intellij) for how to configure IDE settings.
```

**Review by lucasvaltl — APPROVED** [review]
LGTM, made some wording suggestions

File: src/lib/contents/changelog/2022-11-30.md (hunk 9 above)
Comment (atduarte) [reply src/lib/contents/changelog/2022-11-30.md:40]: Slightly modified suggestion:
```suggestion
You can now configure your IDE settings at the project and user level. At the [Project](/docs/configure/projects) level, you can use your [Workspace Image](/docs/configure/workspaces/workspace-image) or commit settings to Git. At the user level, you can now configure settings by leveraging [Dotfiles](/docs/configure/user-settings/dotfiles).
```

**Comment by jldec** [comment]
Please don't merge this - I'm working on adding the WebApp parts

**Review by andreafalzetti — APPROVED** [review]

**Comment by jldec** [comment]
Ready to merge now.

**Pull request merged**